internal/services/workloads/workloads_sap_deployment_virtual_instance_resource.go

```diff
@@ -0,0 +1,2397 @@
+}
+
+type FullResourceNames struct {
+	ApplicationServer []ApplicationServerFullResourceNames `tfschema:"application_server"`
+	CentralServer     []CentralServerFullResourceNames     `tfschema:"central_server"`
+	DatabaseServer    []DatabaseServerFullResourceNames    `tfschema:"database_server"`
+	SharedStorage     []SharedStorage                      `tfschema:"shared_storage"`
+}
+
+type ApplicationServerFullResourceNames struct {
+	AvailabilitySetName string                            `tfschema:"availability_set_name"`
+	VirtualMachines     []VirtualMachineFullResourceNames `tfschema:"virtual_machine"`
+}
+
+type CentralServerFullResourceNames struct {
+	AvailabilitySetName string                            `tfschema:"availability_set_name"`
+	LoadBalancer        []LoadBalancer                    `tfschema:"load_balancer"`
+	VirtualMachines     []VirtualMachineFullResourceNames `tfschema:"virtual_machine"`
+}
+
+type DatabaseServerFullResourceNames struct {
+	AvailabilitySetName string                            `tfschema:"availability_set_name"`
+	LoadBalancer        []LoadBalancer                    `tfschema:"load_balancer"`
+	VirtualMachines     []VirtualMachineFullResourceNames `tfschema:"virtual_machine"`
+}
+
+type LoadBalancer struct {
+	BackendPoolNames             []string `tfschema:"backend_pool_names"`
+	FrontendIpConfigurationNames []string `tfschema:"frontend_ip_configuration_names"`
+	HealthProbeNames             []string `tfschema:"health_probe_names"`
+	Name                         string   `tfschema:"name"`
+}
+
+type SharedStorage struct {
+	AccountName         string `tfschema:"account_name"`
+	PrivateEndpointName string `tfschema:"private_endpoint_name"`
+}
+
+type WorkloadsSAPDeploymentVirtualInstanceResource struct{}
+
+var _ sdk.ResourceWithUpdate = WorkloadsSAPDeploymentVirtualInstanceResource{}
+
+func (r WorkloadsSAPDeploymentVirtualInstanceResource) ResourceType() string {
+	return "azurerm_workloads_sap_deployment_virtual_instance"
+}
+
+func (r WorkloadsSAPDeploymentVirtualInstanceResource) ModelObject() interface{} {
+	return &WorkloadsSAPDeploymentVirtualInstanceModel{}
+}
+
+func (r WorkloadsSAPDeploymentVirtualInstanceResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return sapvirtualinstances.ValidateSapVirtualInstanceID
+}
+
+func (r WorkloadsSAPDeploymentVirtualInstanceResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.SAPVirtualInstanceName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"deployment_with_os_configuration": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
```

Comment:
why `softwareConfiguration` is not defined in the schema?

Reply:
Service team confirmed that we don't need to support it since it cost much time to deploy so that it's not proper to be implemented in TF.